source/mqtt_agent.c

```diff
@@ -515,18 +555,8 @@ static MQTTStatus_t processCommand( MQTTAgentContext_t * pMqttAgentContext,
         commandOutParams.addAcknowledgment &&
         ( commandOutParams.packetId != MQTT_PACKET_ID_INVALID ) )
     {
-        ackAdded = addAwaitingOperation( pMqttAgentContext, commandOutParams.packetId, pCommand );
-
-        /* Set the return status if no memory was available to store the operation
-         * information. */
-        if( !ackAdded )
-        {
-            LogError( ( "No memory to wait for acknowledgment for packet %u\n", commandOutParams.packetId ) );
-
-            /* All operations that can wait for acks (publish, subscribe,
-             * unsubscribe) require a context. */
-            operationStatus = MQTTNoMemory;
-        }
+        operationStatus = addAwaitingOperation( pMqttAgentContext, commandOutParams.packetId, pCommand );
+        ackAdded = ( operationStatus == MQTTSuccess ) ? true : false;
```

Comment:
The last part isn't really necessary
```suggestion
        ackAdded = ( operationStatus == MQTTSuccess );
```